common/scala/src/main/scala/whisk/connector/kafka/KafkaMessagingProvider.scala

```diff
@@ -101,4 +101,24 @@ object KafkaConfiguration {
   def configMapToKafkaConfig(configMap: Map[String, String]): Map[String, String] = configMap.map {
     case (key, value) => configToKafkaKey(key) -> value
   }
+
+  /**
+   * Prints a warning for each unknown configuration item and returns false if at least one item is unknown.
+   *
+   * @param config the config to be checked
+   * @param validKeys known valid keys to configure
+   * @return true if all configuration keys are known, false if at least one is unknown
+   */
+  def verifyConfig(config: Map[String, String], validKeys: Set[String])(implicit logging: Logging): Boolean = {
+    val passedKeys = config.keySet
+    val knownKeys = validKeys intersect passedKeys
+    val unknownSettings = config -- knownKeys
+
+    if (unknownSettings.nonEmpty) {
+      logging.warn(this, s"potential misconfiguration, unknown settings: ${unknownSettings.mkString(",")}")
```

Comment:
May be just log key names as value may potentially contain sensitive info

Reply:
Sounds reasonable.